File: merger.js
//Question 1
var first_names = ["Preston", "Piper", "Nick", "Conrad", "Arthur"]
var last_names = ["Garvey", "Wright", "Valentine", "Kellogg", "Maxson"]

//Part a
function mergeHandler(arr1, arr2, func){
    if(arr1.length != arr2.length)
        return "Arrays are not the same length";
    return func(arr1, arr2);
}

//Part b
function merge2Single(arr1, arr2){
    var mergedArr = [];
    for(var i = 0; i < arr1.length; i++){
        mergedArr.push(arr1[i] + " " + arr2[i]);
    }
    return mergedArr;
}

//Part c
function merge2Object(arr1, arr2){
    var mergedArr = [];
    for(var i = 0; i < arr1.length; i++){
        var obj = {
            "firstname" : arr1[i],
            "lastname" : arr2[i]
        };
        mergedArr.push(obj);
    }
    return mergedArr;
}

function test(){
    console.log(mergeHandler(first_names, last_names, merge2Object));
}

test();

File: merger.py
#Question 1
first_names = ["Preston", "Piper", "Nick", "Conrad", "Arthur"]
last_names = ["Garvey", "Wright", "Valentine", "Kellogg", "Maxson"]

#Part a
def mergeHandler(arr1, arr2, func):
    if(len(arr1) != len(arr1)):
        print("Arrays are not the same length")
        return

    mergedArr = []
    for i in range(0, len(arr1)):
        mergedArr.append(func(arr1[i], arr2[i]))

    return mergedArr

#Part b
def merge2Single(element1, element2):
    return element1 + " " + element2

#Part c
def merge2Object(element1, element2):
    obj = {
        "firstname" : element1,
        "lastname" : element2
    }
    return obj